proxy/src/project.rs

```diff
@@ -125,6 +125,8 @@ where
     credential: Credential,
     /// The project URN.
     urn: coco::Urn,
+    /// The directory that will be created in path where the git repository will be checked out
+    directory_name: String,
```

Comment:
I think at this point we should just pass the `Project` in. Because that way we can get the `urn`, calculate the directory name, and get the branch`.